log.go

```diff
@@ -99,18 +99,23 @@ func init() {
 	addSubLogger("PRNF", peernotifier.UseLogger)
 	addSubLogger("CHFD", chanfunding.UseLogger)
 
-	addSubLogger(routing.Subsystem, routing.UseLogger)
-	addSubLogger(routing.Subsystem, localchans.UseLogger)
+	addSubLogger(routing.Subsystem, routing.UseLogger, localchans.UseLogger)
 	addSubLogger(routerrpc.Subsystem, routerrpc.UseLogger)
 	addSubLogger(wtclientrpc.Subsystem, wtclientrpc.UseLogger)
 	addSubLogger(chanfitness.Subsystem, chanfitness.UseLogger)
 }
 
 // addSubLogger is a helper method to conveniently create and register the
-// logger of a sub system.
-func addSubLogger(subsystem string, useLogger func(btclog.Logger)) {
+// logger of one or more sub systems.
+func addSubLogger(subsystem string, useLoggers ...func(btclog.Logger)) {
+	// Create and register just a single logger to prevent them from
+	// overwriting each other internally.
 	logger := build.NewSubLogger(subsystem, logWriter.GenSubLogger)
-	setSubLogger(subsystem, logger, useLogger)
+	logWriter.RegisterSubLogger(subsystem, logger)
```

Comment:
To avoid this code duplication, couldn't we also extend `setSubLogger` to take multiple loggers and then call that from here? The `range` can just replace this nil check which is also nice.

Reply:
Good suggestion, fixed